include/sempr/entity/spatial/Geometry.hpp

```diff
@@ -49,36 +69,44 @@ class Geometry : public Entity {
     */
     void transformToCS(SpatialReference::Ptr cs);
 
-    /** Exception that may be thrown during transformToCS. */
-    class TransformException : public std::exception {
-    private:
-        std::string message_;
-    public:
-        explicit TransformException(const std::string& m) : message_(m) {}
-        virtual const char* what() const throw() {
-            return message_.c_str();
-        }
-    };
+    // apply a read only filter to find the min and max values of the geometry in his local coordinate system in 3d.
+    void findEnvelope(geom::Coordinate& min, geom::Coordinate& max);
 
     /**
         Get a new entity with the same geometry (copy) referring to the same instance of
         SpatialReference. (implemented by raw_clone)
     */
     Geometry::Ptr clone() const;
 
-private:
+    static geom::Geometry* importFromWKB(const std::basic_string<char> &buffer);
+    static std::basic_string<char> exportToWKB(const geom::Geometry *geom);
+
+    static geom::Geometry* importFromWKT(const std::string &text);
```

Comment:
They are static and provide the internal geometry of the sempr::entity::Geometry class.
At least the traits need this function and I did not know a better place to provided than because geos will not provide it in such an easy way.

Reply:
I understand -- the alternative to make these functions object methods (and thus encapsulate this functionality further in the entity) is not an option since the traits do not have entity-object to use.
Consider this closed. ;)